artview/components/nav.py

```diff
@@ -63,19 +63,18 @@ def __init__(self, Vradar=None, Vgrid=None, Vfilelist=None,
         else:
             self.Vgrid = Vgrid
         if Vfilelist is None:
-            self.Vfilelist = Variable(0)
+            self.Vfilelist = Variable([])
         else:
             self.Vfilelist = Vfilelist
+        self.filename = ""
+        self.fileindex = 0
 
-        self.sharedVariables = {"Vradar": None,
-                                "Vgrid": None,
-                                "Vfilelist": None}
+        self.sharedVariables = {"Vradar": self.NewFile,
```

Comment:
we display the "current file", so need to update it. (I will make more comment on that in the end)

Also notice, this is our first example of a `New*` function that work for several variables.